syntax = "proto3";

enum ServerType {
  RELAY = 0;
  NBB = 1;
}


message RequestServer{
  ServerType server_type = 1;
  string site_code = 2;
  string hardware_uuid = 3;
}

message Server{
  int64 id = 1;  /* server_id */
  string name = 2;
  string type = 3;
  string hostname = 4;
  string ipaddr = 5;
  string version = 6;
  string os_ver = 7;
  string hardware_uuid = 8;
  string machine_id = 9;
}

message ServerRequest {
  RequestServer server = 1;
  int64 server_id = 2;
}

message ServerResponse {
  repeated Server servers = 1;
}

message RelayServer{
  string hardware_uuid = 1;
  string machine_id = 2;
  string site_code = 3;
  string manufacturer = 4;
  string product_name = 5;
  string serial_number = 6;
}

message RelayServerResponse{
  enum RelayStatus {
    REGISTER = 0;
    NOTFOUND = 1;
  }
  RelayStatus status = 1;
}
